src/lib/storageUtils.js

```diff
@@ -162,13 +162,54 @@ export function writeValue(type, name, value, expInterval, domain) {
  * Delete value from cookies or local storage
  * @param {string} type Storage type
  * @param {string} name Name of the item
+ * @param {string} domain Cookie cookieDomain
  */
-export function deleteValue(type, name) {
+export function deleteValue(type, name, domain) {
     if (name) {
         if (type === COOKIE) {
-            delCookie(name);
+            delCookie(name, domain, '/', 'Lax');
         } else if (type === LOCAL_STORAGE) {
             removeStorageItem(name);
         }
     }
 }
+
+/**
+ * Find the top most domain that a cookie can be stored.
+ * www.example.com => example.com
+ * news.example.co.za => example.co.za
+ *
+ * @param {string} hostname Usually document.location.hostname
+ * @returns {string} domain name
+ */
+export function extractDomain(hostname) {
+    const value = '' + Math.floor(Math.random()*10000);
+    const cookieName = `__dmtester_${value}`
+    const sameSite = 'Lax';
+    const parts = hostname.split('.');
+    let lastDomain;
+
+    for (let i = 0, len = parts.length; i < len; ++i) {
```

Comment:
I've always struggled to come up with a good solution to this problem.  I tried to read through this but couldn't quite understand what you are doing.  Perhaps a few lines of documentation could be helpful